File: docs/src/testers/pages/linkedin_step.py
# @sniptest filename=linkedin_step.py
from notte_sdk import NotteClient

client = NotteClient()

with client.Session() as page:
    url = "https://www.linkedin.com/"

    # observe page and take a step
    page.execute(type="goto", url=url)
    actions = page.observe(instructions="click 'jobs'")
    res = page.execute(actions[0])
    print(res.message)

    # another one
    actions = page.observe(instructions="dismiss the sign in check")
    res = page.execute(actions[0])
    print(res.message)

File: docs/src/testers/pages/linkedin_step.ts
import { NotteClient } from 'notte-sdk';

const client = new NotteClient({
  apiKey: process.env.NOTTE_API_KEY,
});

await client.Session().use(async (page) => {
  const url = 'https://www.linkedin.com/';

  // Observe page and take a step
  await page.execute({ type: 'goto', url });
  let actions = await page.observe({ instructions: "click 'jobs'" });
  let result = await page.execute(actions[0]);
  console.log(result.message);

  // Another one
  actions = await page.observe({ instructions: 'dismiss the sign in check' });
  result = await page.execute(actions[0]);
  console.log(result.message);
});
